tests/pipelines/test_pipelines.py

```diff
@@ -1812,6 +1875,18 @@ def test_wrong_model(self):
         assert "is of type" in str(error_context.exception)
         assert "but should be" in str(error_context.exception)
 
+    @require_hf_hub_version_greater("0.26.5")
+    @require_transformers_version_greater("4.47.1")
+    def test_dduf_load_sharded_checkpoint_diffusion_model(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            pipe = DiffusionPipeline.from_pretrained(
+                "hf-internal-testing/tiny-flux-dev-pipe-sharded-checkpoint-DDUF",
+                dduf_file="tiny-flux-dev-pipe-sharded-checkpoint.dduf",
+                cache_dir=tmpdir,
+            ).to(torch_device)
+
+            pipe(prompt="dog", num_inference_steps=5, generator=torch.manual_seed(0), output_type="np").images
```

Comment:
Here, I would:

1. save the `pipe` and load it.
2. run inference with the same params
3. compare the outputs to what we got from `pipe(prompt="dog", num_inference_steps=5, generator=torch.manual_seed(0), output_type="np").images`.